build_tools/cmake/iree_python.cmake

```diff
@@ -279,6 +279,9 @@ function(iree_py_test)
     "ARGS;LABELS;TIMEOUT"
     ${ARGN}
   )
+  if(NOT IREE_BUILD_PYTHON_BINDINGS)
+    return()
+  endif()
 
   iree_local_py_test(
```

Comment:
Want to also (or instead) skip in `iree_local_py_test` above?
current:
```cmake
function(iree_local_py_test)
  if(NOT IREE_BUILD_TESTS OR ANDROID OR EMSCRIPTEN)
    return()
  endif()
```

new:
```cmake
function(iree_local_py_test)
  if(NOT IREE_BUILD_TESTS OR NOT IREE_BUILD_PYTHON_BINDINGS OR ANDROID OR EMSCRIPTEN)
    return()
  endif()
```

(maybe iOS and other platforms should also be in there, or `IREE_BUILD_PYTHON_BINDINGS` could be a `cmake_dependent_option` that is always off on platforms that don't support Python)

Reply:
I did it here because this is the part that adds a dependency on IREE's Python bindings specifically. `iree_local_py_test` could be used for any Python test.